conjure-java-core/src/main/java/com/palantir/conjure/java/services/ServiceGenerators.java

```diff
@@ -17,45 +17,63 @@
 package com.palantir.conjure.java.services;
 
 import com.google.common.base.Strings;
+import com.palantir.conjure.java.util.ErrorGenerationUtils;
 import com.palantir.conjure.java.util.Javadoc;
+import com.palantir.conjure.java.util.Packages;
 import com.palantir.conjure.spec.EndpointDefinition;
+import com.palantir.javapoet.ClassName;
+import com.palantir.javapoet.CodeBlock;
+import com.palantir.javapoet.MethodSpec;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 public final class ServiceGenerators {
+    public record EndpointJavaDocGenerationOptions(boolean includeRequestLine, boolean includeEndpointErrors) {}
 
-    public static Optional<String> getJavaDoc(EndpointDefinition endpointDef) {
-        return getJavaDocInternal(endpointDef, false);
+    public static void addJavaDocForEndpointDefinition(
```

Comment:
Introduced this method which adds the javadoc to the provided `methodBuilder`. I opted to add an "options" record because there are two dimensions of generation options we need to support: 1) if we should include the request line, 2) if we should include endpoint errors.

| Usage | Request Line? | Endpoint Errors? | 
| --- | --- | --- |
| Jersey service generation | No | No |
| Undertow service generation | Yes | Yes |
| Dialogue interface generation | Yes | No* |

*Not yet at least. We can either decide to return a result type in Dialogue, or throw a checked exception. If we do the latter, we need to add `@throws` pieces to the JavaDoc similar to what we do when generating Undertow service interfaces. If we go with the former, we'll still need to document the result union type but the JavaDoc would be added where we define the results type.

Reply:
Ah right, sorry, missed the piece about dialogue, definitely don't want to leak the server-side piece there!